jax/experimental/checkify.py

```diff
@@ -298,6 +301,39 @@ def scan_error_check(error, *in_flat, reverse, length, jaxpr, num_consts, num_ca
   return outs, Error(err, code, new_msgs)
 error_checks[control_flow.scan_p] = scan_error_check
 
+def checkify_while_body_jaxpr(cond_jaxpr, body_jaxpr, error):
+  cond_f = core.jaxpr_as_fun(cond_jaxpr)
+  body_f = core.jaxpr_as_fun(body_jaxpr)
+  def new_body_f(*vals):
+    _ = cond_f(*vals)
+    return body_f(*vals)
+  return checkify_fun_to_jaxpr(lu.wrap_init(new_body_f), error, body_jaxpr.in_avals)
+
+def ignore_errors_jaxpr(jaxpr, error):
+  """Constructs a jaxpr which takes two extra args but ignores them."""
+  f = core.jaxpr_as_fun(jaxpr)
```

Comment:
I think it might be safe to implement this function as roughly

```python
  err_aval = ...
  code_aval = ...
  new_jaxpr = core.Jaxpr(jaxpr.jaxpr.constvars, (core.Var(err_aval), core.Var(code_aval), *jaxpr.jaxpr.invars), ...)
  return core.ClosedJaxpr(jaxpr.consts, new_jaxpr)
```

That would save the tracing round trip.

Reply:
Cool, I've done that!

I used `core.gensym` to generate Vars which do not conflict with the existing Jaxpr Vars, not sure if that's the best way.